apps/dashboard/tests/page-models/create-workflow-sidebar.ts

```diff
@@ -0,0 +1,70 @@
+import { expect, type Page } from '@playwright/test';
+
+export class CreateWorkflowSidebar {
+  constructor(private page: Page) {}
+
+  async createBtnClick({ awaitResponse = false } = {}): Promise<void> {
+    const createWorkflowBtn = await this.page.getByRole('button', { name: 'Create workflow' });
+    await createWorkflowBtn.click();
+
+    if (awaitResponse) {
+      await this.page.waitForResponse(
+        (resp) => resp.url().includes('/v2/workflows') && resp.request().method() === 'POST' && resp.status() === 201
+      );
+    }
+  }
+
+  async expectNameValidationError(): Promise<void> {
+    await expect(await this.page.getByText('Name is required')).toBeVisible();
+  }
+
+  async expectTagsValidationError(): Promise<void> {
+    await expect(await this.page.getByText('Tags must contain at most 16')).toBeVisible();
+  }
+
+  async fillForm({
+    workflowName,
+    workflowId,
+    workflowDescription,
+    tags,
+  }: {
+    workflowName: string;
+    workflowId: string;
+    workflowDescription: string;
+    tags: string[] | number;
+  }): Promise<void> {
+    // fill the workflow name
+    await this.page.locator('input[name="name"]').fill(workflowName);
+    const workflowIdInput = await this.page.locator('input[name="workflowId"]');
+
+    // check the workflow id
+    await expect(await workflowIdInput.inputValue()).toEqual(workflowId);
```

Comment:
Following on my previous comment, this is a good example of overloading a page object function with expectation logic that belongs to the test.